src/utils/error.ts

```diff
@@ -0,0 +1,35 @@
+import { AxiosError } from 'axios';
+import { authenticateUser } from '../auth/ducks/actions';
+
+export interface ApiError {
+  readonly message: string;
+  readonly code?: string;
+  readonly type: ApiErrorTypes;
+}
+
+export enum ApiErrorTypes {
+  NETWORK_ERROR = 'NETWORK_ERROR',
+  GENERIC_API_ERROR = 'GENERIC_API_ERROR',
+}
+
+export const axiosErrorToApiError = (error: AxiosError): ApiError => {
+  if (error.response) {
+    return {
+      message: error.response.data,
+      code: error.code,
+      type: ApiErrorTypes.GENERIC_API_ERROR,
+    };
+  } else if (error.request) {
+    return {
+      message: error.message,
+      code: error.code,
+      type: ApiErrorTypes.NETWORK_ERROR,
+    };
+  } else {
+    return {
+      message: error.message,
+      code: error.code,
+      type: ApiErrorTypes.GENERIC_API_ERROR,
```

Comment:
This should have its own type since if we are in this block, the error was when set up the request. Maybe like REQUEST_ERROR? It is true that if we have no bugs we should never reach here in production build